### ISSUE TITLE
meta_value questions

### ISSUE THREAD

**Issue by laubeke** [microsoft/AutoML#25]
Hi,

thank You for Your quick responses so far, I hope You can bear with me.
I have two questions regarding the following piece of code (see https://github.com/microsoft/Cream/blob/main/lib/models/PrioritizedBoard.py#L27)

```
        elif self.cfg.SUPERNET.PICK_METHOD == 'meta':
            meta_value, cand_idx, teacher_cand = -1000000000, -1, None
            for now_idx, item in enumerate(self.prioritized_board):
                inputx = item[4]
                output = F.softmax(model(inputx, random_cand), dim=1)
                weight = model.module.forward_meta(output - item[5])
                if weight > meta_value:
                    meta_value = weight
                    cand_idx = now_idx
                    teacher_cand = self.prioritized_board[cand_idx][3]
            assert teacher_cand is not None
            meta_value = torch.sigmoid(-weight)
    ...
    return meta_value, teacher_cand
```

1) why use `torch.sigmoid(-weight)`, and not `torch.sigmoid(-meta_value)` ?
`weight` will refer to the last entry in the board, and might not belong to the selected `teacher_cand`

2) why use `torch.sigmoid(-weight)`, and not `torch.sigmoid(weight)` ?
since `weight` describes the matching degree (higher is better, the teacher candidate is also selected this way), `torch.sigmoid(-weight)` will go towards 0.
subsequently, a high matching value has lower L_kd impact on the loss function (https://github.com/microsoft/Cream/blob/main/lib/core/train.py#L64):
`loss = (meta_value * kd_loss + (2 - meta_value) * valid_loss) / 2`

Am I missing something?

Best,
Kevin

**Comment by Z7zuqer** [comment]
Hi,

Thanks for your interest in our project! It should be `torch.sigmoid(meta_value)`. This code is extracted from a large codebase, this is caused by carelessness. We would fix it in a few days.

Best,

Hao.

**Comment by sunnyxiaohu** [comment]
Hi,
Thanks for your great job!
[the way to simulate sgd updating. ](https://github.com/microsoft/AutoML/blob/main/Cream/lib/models/MetaMatchingNetwork.py#L45)
```
  # simulate sgd updating
  def simulate_sgd_update(self, w, g, optimizer):
      return g * optimizer.param_groups[-1]['lr'] + w
```
According to Eq(6) in the paper, It should be ``return -g * optimizer.param_groups[-1]['lr'] + w`` ?

**Comment by Z7zuqer** [comment]
Hi,

Thanks for your interest in our project!

It seems that when we were re-organizing the codebase, we made some mistakes. We will fix it ASAP, and thanks for pointing it out!

Best,

Hao.
